storage/allocator.go

```diff
@@ -41,6 +42,10 @@ const (
 	// is within rebalanceFromMean of the mean, it is considered a
 	// viable target to rebalance to.
 	rebalanceFromMean = 0.025 // 2.5%
+	// rebalanceShouldRebalanceChance represents a chance that an individual
```

Comment:
It's free, but it's also limiting: each range has an equal chance of getting selected. A separate rebalancing thread would be able to make other choices, such as favoring inactive ranges to minimize disruption (unless we're rebalancing for QPS), or more generally weighting ranges by how much they contribute to the metric we're trying to balance.